File: go/assets/encrypt.go
package assets

import (
	"fmt"
	"strconv"
)

func encryptTable() map[int]string {
	emojis := getEmojis()
	i2e := make(map[int]string)
	currentNum := 0
	for _, emoji := range emojis {
		i2e[int(currentNum)] = string(emoji)
		currentNum++
	}
	return i2e
}

func encryptSplit(binaryStream string) []string {
	var bins []string
	current := ""

	for _, i := range binaryStream {
		if len(current) < 7 {
			current += string(i)
		} else {
			bins = append(bins, current)
			current = string(i)
		}
	}
	for len(current) < 7 {
		current += "0"
	}
	bins = append(bins, current)
	return bins
}

func encryptGetStream(numbers []int) (stream string) {
	for _, i := range numbers {
		stream += fmt.Sprintf("%0.08b", int(i))
	}
	return
}

func Encode(message string) string {
	var numbers []int

	for _, i := range message {
		numbers = append(numbers, int(i))
	}
	binaryStream := encryptGetStream(numbers)
	bins := encryptSplit(binaryStream)

	emojis := ""
	i2e := encryptTable()
	for _, bin := range bins {
		i, _ := strconv.ParseInt(string(bin), 2, 64)
		emojis += i2e[int(i)]
	}

	return emojis
}

File: docs/js/encrypt.js
// module "encrypt.mjs"
function getEncryptTable() {
    let emojis = "😀😃😄😁😆😅🤣😂🙂🙃😉😊😇🥰😍🤩😘😗😚😙😋😛😜🤪😝🤑🤗🤭🤫🤔🤐🤨😐😑😶😏😒🙄😬🤥😌😔😪🤤😴😷🤒🤕🤢🤮🤧🥵🥶🥴😵🤯🤠🥳😎🤓🧐😕😟🙁😮😯😲😳🥺😦😧😨😰😥😢😭😱😖😣😞😓😩😫🥱😤😡😠🤬😈👿💀💩🤡👹👺👻👽👾🤖😺😸😹😻😼😽🙀😿😾💋👋🤚🖐✋🖖👌🤏🤞🤟🤘🤙👈👉👆🖕👇👍👎✊"
    let i2e = new Map()
    let index = 0
    for(var emoji of emojis) {
        i2e.set(index, emoji)
        index++
    }
    return i2e
}

function split(binaryStream) {
    let bins = []
    let current = ""

    for(let index in binaryStream) {
        if(current.length < 7){
            current += binaryStream[index]
        } else {
            bins.push(current)
            current = binaryStream[index]
        }
    }

    while(current.length < 7) {
        current += "0"
    }
    bins.push(current)
    return bins
}

function convertToByte(number) {
    bin = number.toString(2)
    if(bin.length > 8) {
        return;
    } else {
        while(bin.length < 8) {
            bin = "0" + bin
        }
    }
    return bin
}

function encryptText(message) {
    let binaryStream = ""
    for(let index in message) {
        let number = message.charCodeAt(index)
        console.log(number)
        binaryStream += convertToByte(number)
    }
    bins = split(binaryStream)
    let emojis = ""
    let i2e = getEncryptTable()
    for(var index in bins) {
        let number = parseInt(bins[index], 2)
        emojis += i2e.get(number)
    }
    return emojis
}